pctiler/pctiler/endpoints/zarr.py

```diff
@@ -0,0 +1,225 @@
+from dataclasses import dataclass
+from typing import List, Literal, Optional, Tuple, Type
+from urllib.parse import urlencode
+
+import xarray
+from fastapi import Depends, Path, Query
+from rio_tiler.io import BaseReader, XarrayReader
+from rio_tiler.models import Info
+from starlette.requests import Request
+from starlette.responses import Response
+from titiler.core.dependencies import RescalingParams
+from titiler.core.factory import BaseTilerFactory, img_endpoint_params
+from titiler.core.models.mapbox import TileJSON
+from titiler.core.resources.enums import ImageType
+from titiler.core.resources.responses import JSONResponse
+
+
+@dataclass
+class XarrayTilerFactory(BaseTilerFactory):
+
+    # Default reader is set to rio_tiler.io.Reader
+    reader: Type[BaseReader] = XarrayReader
+
+    def register_routes(self):
+        """Register Info / Tiles / TileJSON endoints."""
+
+        @self.router.get(
+            "/info",
+            response_model=Info,
+            response_model_exclude_none=True,
+            response_class=JSONResponse,
+            responses={200: {"description": "Return dataset's basic info."}},
+        )
+        def info_endpoint(
+            src_path=Depends(self.path_dependency),
+            variable: str = Query(..., description="Xarray Variable"),
+        ):
+            """Return dataset's basic info."""
+            with xarray.open_dataset(
+                src_path, engine="zarr", decode_coords="all"
+            ) as src:
+                ds = src[variable][:1]
```

Comment:
For now we select the first dimension

Reply:
I added an option to select a time slice (I'm assuming the third dimension will be time)